### ISSUE TITLE
[labs/virtualizer] Using `overflow: auto` by default can trigger a11y failures in axe-core

### ISSUE THREAD

**Issue by Westbrook** [lit/lit#3238]
### Which package(s) are affected?

Virtualizer (@lit-labs/virtualizer)

### Description

By default, when setting content to `overflow: auto` `axe-core` accessibility tests expect that the same element also be tab focusable, e.g. have a `tabindex`. Should Virtualizer indiscriminately apply `overflow: auto` as an inline style without also managing the `tabindex` in someway? Applied inline, consumers are required to leverage `!important` to be able to remove this rule. 

### Reproduction

See this demo: https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCB7dmlydHVhbGl6ZX0gZnJvbSAnQGxpdC1sYWJzL3ZpcnR1YWxpemVyL3ZpcnR1YWxpemUuanMnO1xuXG5AY3VzdG9tRWxlbWVudCgnc2ltcGxlLWdyZWV0aW5nJylcbmV4cG9ydCBjbGFzcyBTaW1wbGVHcmVldGluZyBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgICAgOmhvc3QgeyBcbiAgICAgICAgZGlzcGxheTogYmxvY2s7XG4gICAgICAgIGhlaWdodDogMjAwcHg7XG4gICAgICB9YDtcblxuICBAcHJvcGVydHkoKSBcbiAgcHVibGljIHNlbGVjdGVkID0gbmV3IFNldChbMiwgNV0pO1xuICBcbiAgQHByb3BlcnR5KClcbiAgcHVibGljIGl0ZW1zID0gQXJyYXkuZnJvbShBcnJheSg1KS5rZXlzKCkpO1xuICBcbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYFxuICAgICA8ZGl2PlxuICAgICAgJHt2aXJ0dWFsaXplKHtcbiAgICAgICAgc2Nyb2xsZXI6IHRydWUsIFxuICAgICAgICBpdGVtczogdGhpcy5pdGVtcywgXG4gICAgICAgIHJlbmRlckl0ZW06IChpdGVtKSA9PiBodG1sYDxkaXY-JHtpdGVtfSAke3RoaXMuc2VsZWN0ZWQuaGFzKGl0ZW0pID8gJ3NlbGVjdGVkJyA6ICcnfTwvZGl2PmBcbiAgICAgIH0pfVxuICAgICA8L2Rpdj5cbiAgICBgO1xuICB9XG59In0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL290aGVyLWdyZWV0aW5nLmpzXCI-PC9zY3JpcHQ-XG4gIFxuPC9oZWFkPlxuPGJvZHk-XG4gIDxzaW1wbGUtZ3JlZXRpbmc-PC9zaW1wbGUtZ3JlZXRpbmc-XG48L2JvZHk-In0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ

### Workaround

Do some management of the tab index with a Resize Observer or the like and the hang `overlay: auto` on the presence of `[tabindex]` and then add `overlay: visible !important` when `[tabindex]` is not available.

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

"@lit-labs/virtualizer": "^0.7.1",

### Browser/OS/Node environment

It does this work in all browsers, but Chrome is the most finicky about it at test time in association with `axe-core`.

**Comment by Westbrook** [comment]
cc: @najikahalsema

**Comment by Westbrook** [comment]
Actually, turns out this _wasn't_ the root of the issue...
